dev/ci/shared/runner.py

```
@@ -57,9 +63,10 @@ def setup_cluster(self, args, extra_push_images_args=None):
         if result.returncode != 0:
             return result
 
-        result = subprocess.run([f"{self.repo_root}/dev/tools/deploy-to-kube", "--image-prefix", args.image_prefix, "--image-tag", image_tag, "--extensions"])
-        if result.returncode != 0:
-            return result
+        if not getattr(args, "skip_initial_deploy", False):
+            result = subprocess.run([f"{self.repo_root}/dev/tools/deploy-to-kube", "--image-prefix", args.image_prefix, "--image-tag", image_tag, "--extensions"])
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)